### PR TITLE
Bug in ExtrinsicDeformation

### PR DESCRIPTION
**Pull request by Louis-Pujol**
The current implementation of `ExtrinsicDeformation.morph(shape, parameter)` leads to error if `parameter.requires_grad = True` and parameter is not a leaf variable.

```python
model = sks.ExtrinsicDeformation(
      n_steps=2,
      kernel="gaussian",
      scale=0.5,
  )

  mesh = sks.Sphere().decimate(n_points=20)

  x = torch.rand(3)
  x.requires_grad = True
  parameter_shape = model.parameter_shape(shape=mesh)
  parameter = x.repeat(parameter_shape[0]).reshape(parameter_shape)

 model.morph(
      shape=mesh,
      parameter=parameter,
  )
```

To solve this, it suffices to call `parameter.requires_grad = True` only if it is not already the case. A test is added with a non-leaf variable as parameter to check that the execution does not lead to error, and check that the result of the morphing is consistent w.r.t. the same where the parameter does not requires `grad`.

### CODE REVIEW

**Comment by Louis-Pujol** [comment]
The function `skshapes.morphing.validation.validate_polydata_morphing_model` has been modified to systematically test gradient computation when the parameter of `morph` is a non-leaf variable.
